src/riak_kv_get_core.erl

```diff
@@ -284,6 +288,36 @@ num_pr(GetCore = #getcore{num_pok=NumPOK, idx_type=IdxType}, Idx) ->
             GetCore
     end.
 
+%% @private Print a warning if objects are not equal. Only called on case of no read-repair
+%% This situation could happen with pre 2.1 vclocks in very rare cases. Fixing the object
+%% requires the user to rewrite the object in 2.1+ of Riak. Logic is disabled when using
+%% a defined version of aae hashing as it's no longer needed.
+maybe_log_old_vclock(Results) ->
+    case riak_core_capability:get({riak_kv, object_hash_version}, undefined) of
```

Comment:
So I think the comment is a bit misleading and I have changed it for clarity. We only want to trigger this heavy riak_object comparison and logging when we are in the process of the hashtree upgrade. So, if the user has manually overridden the capability to set `undefined` version, we want this logic off.  If the capability has been set to 0(all nodes are upgraded to 2.2) and the riak_kv_entropy_manager:get_version call returns undefined(anything but 0) then we know the upgrade has not yet finished and we should log in specific situations where we have multiple replica objects in the Result set and they are also not equal.